presets/presetManager.py

```diff
@@ -0,0 +1,223 @@
+import os.path
+import logging
+import json
+
+logger = logging.getLogger(__name__)
+
+
+class PresetManager(object):
+    MAX_DEPTH = 2
+
+    def __init__(self, paths, strict=False):
+        self.presets = {}
+        self.strict = strict
+
+        # normalize paramter type string -> array
+        if isinstance(paths, basestring):
+            paths = [paths]
+
+        self._loadPaths(paths)
+
+    def _loadPaths(self, paths, depth=0):
+        if depth > self.MAX_DEPTH:
+            return
+
+        for i in range(len(paths)):
+            path = paths[i]
+
+            # avoid empty string
+            if not path:
+                continue
+
+            # cleanUp path
+            if depth == 0:
+                path = os.path.expanduser(path)  # replace ~
+                path = os.path.expandvars(path)  # replace vars
+                path = os.path.normpath(path)  # replace /../ and so on
```

Comment:
I don't think that this is true. `normpath` will resolve symbolic links, so if you have a symbolic link inside the "base" directory, normpath will change it. Anyway, this doesn't make so much of a difference, and you could want to leave it this way. But it should at least be commented, clarifying to the reader why is there  such an exception, the fact that it is just an optimization and that there is nothing special about `depth == 0`